src/lib/components/PrismicImage.svelte

```diff
@@ -1,34 +1,63 @@
 <script lang="ts">
-	import * as prismicT from "@prismicio/types";
+	import * as prismicH from "@prismicio/helpers"
+	import type * as prismicT from "@prismicio/types";
+	
+	/**
+	 * A Prismic Image field.
+	 */
 	export let field: prismicT.ImageField;
-	let className: string;
 
-	// creates a `class` property, even
-	// though it is a reserved word
-	export { className as class };
+	/**
+	 * An object of Imgix URL API parameters to transform the image.
+	 *
+	 * See: https://docs.imgix.com/apis/rendering
+	 */
+	export let imgixParams: Parameters<typeof prismicH.asImageSrc>[1] | null | undefined = null;
 
-	const usePrismicImage = (
-		field: prismicT.ImageField,
-	): {
-		url: string;
-		width?: number;
-		height?: number;
-		alt?: string;
-		copyright: string | undefined;
-	} => {
-		field.url;
-		return {
-			url: field.url || "",
-			alt: field.alt || "",
-			width: field.dimensions?.width || undefined,
-			height: field.dimensions?.height || undefined,
-			copyright: field.copyright || undefined,
-		};
-	};
+	/**
+	 * Widths used to build a `srcset` value for the Image field.
+	 *
+	 * If a `widths` prop is not given or `"defaults"` is passed, the
+	 * following widths will be used: 640, 750, 828, 1080, 1200, 1920, 2048,
+	 * 3840.
+	 *
+	 * If the Image field contains responsive views, each responsive view
+	 * can be used as a width in the resulting `srcset` by passing
+	 * `"thumbnails"` as the `widths` prop.
+	 */
+	export let widths: NonNullable<Parameters<typeof prismicH.asImageWidthSrcSet>[1]>["widths"] | null | undefined = null;
+				
+	
+	/**
+	 * HTML class attribute. Use a `:global()` selector to target the class.
+	 */
+	let className: string | null | undefined = null;
+	export { className as class };
 
-	const image = usePrismicImage(field);
+	const params = {}
+	if(widths) Object.assign(params, { widths })
+	if(imgixParams) Object.assign(params, imgixParams)
 </script>
 
-{#if field.url}
-	<img src={image.url} alt={image.alt} height={image.height} />
-{/if}
+<!-- 
+  @component
+  Component to render a Prismic Image field as an `img` element with `width`, `height`, `alt`, `src`, and `srcset` attributes.
+  
+  @example Rendering an Image field:
+	```svelte
+		<PrismicImage 
+			field={document.data.example_image} 
+			imgixParams={{ sat: -100 }}
+		/>
+  ```
+-->
+
+{#if prismicH.isFilled.image(field)}
+	{@const { alt, dimensions } = field}
+	{@const { width, height } = dimensions }
+	{@const { src, srcset } = prismicH.asImageWidthSrcSet(field, params)}
+
+	<img {src} {srcset} {alt} {width} {height} class={className} />
+{:else}
+	<img alt={field.alt || ""} class={className} />
+{/if}
```

Comment:
1. `alt` should be overridable to declare the image as decorative.
2. Marking an image as decorative should not be a possible default output.

Details on this behavior in `@prismicio/react` can be found here: https://github.com/prismicio/prismic-react/blob/master/messages/alt-must-be-an-empty-string.md

Could we replicate `@prismicio/react`'s `alt` handling?

Reply:
I think I've recreated this. I put in the console warnings for `alt` and `fallbackAlt`. However, I'm not sure if we can recreate the `__PRODUCTION__` variable in Svelte. Let me know if you think I should try it and I'll do so. In lieu of that, the console warnings will always show.

(SvelteKit has a `dev` variable, but it's not available without SvelteKit.)

Reply:
Today I learned that this behavior is actually okay! `<img alt="" />` is equivalent to `<img alt />`.

> **Empty attribute syntax**
> Just the [attribute name](https://html.spec.whatwg.org/multipage/syntax.html#syntax-attribute-name). The value is implicitly the empty string.

Source: https://html.spec.whatwg.org/multipage/syntax.html#attributes-2

React seems to have the same output as React when you give it `<img alt="" />`. Svelte and React also behave the same way when you provide it `<img alt={null} />` (outputs `<img />`), which is what we want when an Image field's `alt` property is empty.

Reply:
Also confirmed by Rich Harris: https://twitter.com/Rich_Harris/status/1630293839272345600

I've uncommented the alt prop and functionality.